### PR TITLE
Adding Full Mexican Holidays

### PR DESCRIPTION
**Pull request by davsaniuv**
im adding the full list of mandatory and non mandatory days  with the recent updates according to the official [government website](https://www.gob.mx/profedet/articulos/sabes-cuales-son-los-dias-de-descanso-obligatorio-para-este-2024)

### CODE REVIEW

**Comment by Nielsvanpach** [comment]
There already seems to be a Mexican PR. Could you guys work on 1 PR together https://github.com/spatie/holidays/pull/20

File: src/Countries/Mexico.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        return 'mx';
+    }
+
+    /** @return array<string, string|CarbonImmutable> */
+    protected function allHolidays(int $year): array
+    {
+
+        $natalicioBenitoJuarez = new CarbonImmutable(sprintf("third monday of march %s", $year));
+        $promulgacionConstitucion = new CarbonImmutable(sprintf("first monday of february %s", $year));
+        $revolucionMexicana = CarbonImmutable::createFromFormat('Y-m-d', "{$year}-11-20")->setTimezone('America/Mexico_City');
```
Comment (duartec2) [inline src/Countries/Mexico.php:40]: **Wrong date!** As per [mexican government website](https://www.gob.mx/profedet/articulos/sabes-cuales-son-los-dias-de-descanso-obligatorio-para-este-2024) "El día de la revolución mexicana" is the third monday of november

File: src/Countries/Mexico.php [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if ($revolucionMexicana->isSunday()) {
+            $revolucionMexicana = $revolucionMexicana->next('monday');
+        }
+
+        $mandatory = [
+            'Año nuevo' => '01-01',
+            'Aniversario de la promulgación de la Constitución de 1917' => $promulgacionConstitucion->format('m-d'),
+            'Natalicio de Benito Juárez' => $natalicioBenitoJuarez->format('m-d'),
+            'Día del Trabajo' => '05-01',
+            'Día de la Independencia' => '09-15',
```
Comment (duartec2) [inline src/Countries/Mexico.php]: "Día de la independencia" it's on Sept. 16

File: src/Countries/Mexico.php [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    {
+
+        $fathersDay = new CarbonImmutable(sprintf("third sunday of june %s", $year));
+
+        return [
+
+            'Día de la Candelaria' => '02-02',
+            'Día de la Bandera' => '02-24',
+            'Día del Niño' => '04-30',
+            'Día de la Madre' => '04-30',
```
Comment (duartec2) [inline src/Countries/Mexico.php]: "Día de la madre" it's on May 10

**Comment by freekmurze** [comment]
Thanks!

**Comment by davsaniuv** [comment]
I've fixed all in this [PR](https://github.com/spatie/holidays/pull/130)

**Pull request merged**